src/knot_protocol.c

```diff
@@ -172,21 +159,20 @@ int knot_type_id_is_generic(uint16_t type_id)
 
 int knot_schema_is_valid(uint16_t type_id, uint8_t value_type, uint8_t unit)
 {
-	/* int/float/bool/raw ? */
+	/* int/float/bool/raw/int64/uint/uint64 ? */
 	if (knot_value_type_is_valid(value_type) == 0) {
 
 		/* Verify basic type IDs */
 		if (knot_type_id_is_basic(type_id) == 0) {
-			if (basic_types[type_id].value_type == value_type &&
-			    basic_types[type_id].unit_mask & UNIT_MASK(unit))
+			if (basic_types[type_id] & UNIT_MASK(unit))
 				return 0;
 
 		/* Verify logic type IDs */
 		} else if (knot_type_id_is_logic(type_id) == 0) {
-			if (logic_types[TYPE_ID_MASK(type_id)] == value_type)
+			if (unit == KNOT_UNIT_NOT_APPLICABLE)
 				return 0;
 		} else if (knot_type_id_is_generic(type_id) == 0) {
-			if (generic_types[TYPE_ID_MASK(type_id)] == value_type)
+			if (unit == KNOT_UNIT_NOT_APPLICABLE)
 				return 0;
 		}
```

Comment:
Could these checks be merged? They're doing the same thing.

Reply:
Yes, I tried to keep the same code structure since it was representing different groups of type ids, but I can change it if you prefer.